packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```diff
@@ -184,8 +195,29 @@ export function applyAgentEdit(
 			let insertedObject: TreeNode | undefined;
 			// if fieldSchema is a LeafnodeSchema, we can check that it's a valid type and set the field.
 			if (isPrimitive(modification)) {
-				// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
-				(node as any)[treeEdit.field] = modification;
+				try {
+					// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
+					(node as any)[treeEdit.field] = modification;
```

Comment:
I know this was already here, just checking why we need to cast as `any` here... feels like we should be able to use strongly typed APIs.

Reply:
I think this has to do with the fact that treeEdit.field is typed as a string and it is not guaranteed to be a part of the schema of the given tree node

Reply:
I believe we should be able to narrow `node` down to its schema aware typing. @noencke can you advise on whether or not this is correct / what API should be called?

Reply:
Sean's analysis sounds correct to me. We'd have to narrow `node`, `treeEdit.field` _and_ `modification` to their concrete values at compile time for this to work without casts. That said, we can probably still do a little better than `any`. How about `(node as Record<string, typeof modification>)[treeEdit.field] = modification`?

Also, have we already validated that the primitive is the type we expect? We validate that it is a primitive at all (via `isPrimitive`, but do we also validate that it's not a `number` when we expect a `string`? We should. You could add the field schema as an optional second parameter to `isPrimitive`. If present, it checks that the type of the first parameter is allowed in the field.